backend/de.metas.contracts/src/main/java/de/metas/contracts/commission/commissioninstance/businesslogic/sales/commissiontrigger/salesinvoiceline/SalesInvoiceFactory.java

```diff
@@ -143,10 +146,34 @@ public SalesInvoiceFactory(@NonNull final CommissionProductService commissionPro
 					continue;
 				}
 				final List<I_C_Invoice_Candidate> invoiceCandidates = invoiceCandDAO.retrieveIcForIl(invoiceLineRecord);
+
+				boolean isCreditMemoNotReinvoiceable = false;
 				if (!invoiceCandidates.isEmpty())
 				{
-					logger.debug("C_InvoiceLine is not manual as it has {} C_Invoice_Candidates; -> return empty", invoiceLineRecord.getM_Product_ID());
-					continue;
+					if(invoiceIsCreditMemo)
+					{
+
+						for (final I_C_Invoice_Candidate invoiceCandidate : invoiceCandidates)
+						{
+							final I_C_Invoice_Line_Alloc alloc = invoiceCandDAO.retrieveIlaForIcAndIl(invoiceCandidate, invoiceLineRecord);
+
+							final InvoiceLineAllocType invoiceLineAllocType = InvoiceLineAllocType.ofCodeNullable(alloc.getC_Invoice_Line_Alloc_Type());
+
+							if (invoiceLineAllocType == null || !invoiceLineAllocType.IsCreditMemoNotReinvoiceable())
```

Comment:
i think this is right but i had to knot my brain around the ! and `not`
maybe extract a local var?

or a (static) method like "isTypeOfStandaloneCreditMemo(invoiceLineAllocType)" ....idk..something that helps to understand it...